src/ansys/geometry/core/geometry/curves/trimmed_curve.py

```diff
@@ -0,0 +1,209 @@
+# Copyright (C) 2023 - 2024 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""Trimmed curve class."""
+from ansys.api.geometry.v0.commands_pb2 import IntersectCurvesRequest
+from ansys.api.geometry.v0.commands_pb2_grpc import CommandsStub
+from beartype.typing import List
+from pint import Quantity
+
+from ansys.geometry.core.connection.client import GrpcClient
+from ansys.geometry.core.connection.conversions import trimmed_curve_to_grpc_trimmed_curve
+from ansys.geometry.core.errors import protect_grpc
+from ansys.geometry.core.geometry.curves.curve import Curve
+from ansys.geometry.core.geometry.curves.curve_evaluation import CurveEvaluation
+from ansys.geometry.core.geometry.parameterization import Interval
+from ansys.geometry.core.math import Point3D
+from ansys.geometry.core.typing import Real
+
+
+class TrimmedCurve:
+    """
+    Represents a trimmed curve.
+
+    A trimmed curve is a curve that has a boundary. This boundary comes in the form of an Interval.
+
+    Parameters
+    ----------
+    geometry : Curve
+        The underlying mathematical representation of the curve.
+    start : Point3D
+        The start point of the curve.
+    end : Point3D
+        The end point of the curve.
+    interval : Interval
+        The parametric interval that bounds the curve.
+    length : Quantity
+        The length of the curve.
+    grpc_client : GrpcClient, default: None
+        Optional gRPC client that is required for advanced functions such as `intersect_curve()`.
```

Comment:
```suggestion
    geometry : Curve
        Underlying mathematical representation of the curve.
    start : Point3D
        Start point of the curve.
    end : Point3D
        End point of the curve.
    interval : Interval
        Parametric interval that bounds the curve.
    length : Quantity
        Length of the curve.
    grpc_client : GrpcClient, default: None
        gRPC client that is required for advanced functions such as ``intersect_curve()``.
```
Should function be used (is it ouside a class?) or should method be used (is it inside a class)?